src/models/appmodel.ts

```diff
@@ -11,17 +12,39 @@ export class AppModel implements IAppModel {
 
     @observable private _selectedFlavor = "";
     get selectedFlavor(): string { return this._selectedFlavor; }
-    set selectedFlavor(item: string) { this._selectedFlavor = item; }
+    set selectedFlavor(item: string) { 
+        this._selectedFlavor = item; 
+        if(!this._isConstructing) this.saveState();
+    }
 
+    private _isConstructing = true;q
     @observable private _textInput = "";
     get flavorInput(): string { return this._textInput; }
     set flavorInput(value: string) { this._textInput = value; }
 
     get flavorTextIsValid() { return Boolean(this.flavorInput); }
 
-    constructor()
+    private _dataModel: IDataModel;
+
+    constructor(dataModel: IDataModel)
     {
         this.selectedFlavor = this.flavors[0];
+        this._dataModel = dataModel;
+        const data = dataModel.load();
+        if(data != null && data != "")
+        {
+            this.flavors.clear();
+            const jsonData = JSON.parse(data);
```

Comment:
I'd rather move this code to some method, just not to keep everything in the constructor
in the constructor I'd just call this method (eg.   this.init(args) ) etc,

Reply:
Also, I think that it'd be nice to add JSON.parse to the storage layer. It would make our life a bit easier because we wouldn't check strings but rather original types.

Reply:
I started down the path of putting this in the storage layer and it adds a lot of complexity to the app, especially if I try to preserve types.   It also adds extra logic in the storage layer, which is more difficult  to test.   For this demo, I think storing as text is probably the simplest approach.    

If we are going to move the code into an init method, let's call it asynchronously.   I'll make a card for that.

Reply:
Well, we can just create a correct type in our storage _load_ method. What it should return
and we can just return the JSON.parse(data) and our components will receive the ready-to-use json object
I can do it if you want

btw, about Async init it's a good idea